manticore/native/manticore.py

```diff
@@ -0,0 +1,192 @@
+
+from manticore.core.manticore import ManticoreBase
+from manticore import STDIN_INPUT_DEFAULT_SIZE
+
+from manticore.core.smtlib import ConstraintSet
+from manticore.utils import log
+from manticore.utils.helpers import issymbolic
+from .state import State
+
+logger = logging.getLogger(__name__)
+log.init_logging()
+
+
+class Manticore(ManticoreBase):
+    def __init__(self, path_or_state, argv=None, workspace_url=None, policy='random', **kwargs):
+        """
+        :param path_or_state: Path to binary or a state (object) to begin from.
+        :param argv: arguments passed to the binary.
+        """
+        if isinstance(path_or_state, str):
+            if not os.path.isfile(path_or_state):
+                raise OSError(f'{path_or_state} is not an existing regular file')
+
+            initial_state = _make_initial_state(path_or_state, argv=argv, **kwargs)
+        else:
+            initial_state = path_or_state
+
+        super().__init__(initial_state, workspace_url=workspace_url, policy=policy, **kwargs)
+
+    @classmethod
+    def linux(cls, path, argv=None, envp=None, entry_symbol=None, symbolic_files=None, concrete_start='', stdin_size=STDIN_INPUT_DEFAULT_SIZE, **kwargs):
+        """
+        Constructor for Linux binary analysis.
+
+        :param str path: Path to binary to analyze
+        :param argv: Arguments to provide to the binary
+        :type argv: list[str]
+        :param envp: Environment to provide to the binary
+        :type envp: dict[str, str]
+        :param entry_symbol: Entry symbol to resolve to start execution
+        :type envp: str
+        :param symbolic_files: Filenames to mark as having symbolic input
+        :type symbolic_files: list[str]
+        :param str concrete_start: Concrete stdin to use before symbolic input
+        :param int stdin_size: symbolic stdin size to use
+        :param kwargs: Forwarded to the Manticore constructor
+        :return: Manticore instance, initialized with a Linux State
+        :rtype: Manticore
+        """
+        try:
+            return cls(_make_linux(path, argv, envp, entry_symbol, symbolic_files, concrete_start, stdin_size), **kwargs)
+        except elftools.common.exceptions.ELFError:
+            raise Exception(f'Invalid binary: {path}')
+
+    @classmethod
+    def decree(cls, path, concrete_start='', **kwargs):
+        """
+        Constructor for Decree binary analysis.
+
+        :param str path: Path to binary to analyze
+        :param str concrete_start: Concrete stdin to use before symbolic input
+        :param kwargs: Forwarded to the Manticore constructor
+        :return: Manticore instance, initialized with a Decree State
+        :rtype: Manticore
+        """
+        try:
+            return cls(_make_decree(path, concrete_start), **kwargs)
+        except KeyError:  # FIXME(mark) magic parsing for DECREE should raise better error
+            raise Exception(f'Invalid binary: {path}')
+
+    def invoke_model(self, model):
```

Comment:
i think this should be on the native State, rather than on native Manticore

Reply:
Nice catch! I have no idea how I moved it to Manticore :|.